### PR TITLE
Fix inconsistencies when building with CFG_WITH_USER_TA=n

### PR DESCRIPTION
**Pull request by maroueneboubakri**
Here is summary of the issues and improvements:
When `CFG_WITH_USER_TA` is not set:

1. The `copy_from_user() `and` copy_from_user_private()` functions are re-defined in `user_access.c` in addition to `user_access.h`. This leads to compilation errors.

2. The` tee_ta_enc_manager.c` is built regardless `CFG_WITH_USER_TA` is set or not. This increases the footprint with unused functions, therefore, decide to build it or not based on `CFG_WITH_USER_TA`. In addition make sure not to use functions which will not be resolved due to the other sources which depends already on  `CFG_WITH_USER_TA`.

3. The default `tee_otp_get_ta_enc_key()` function references `ta_pub_key_modulus_size` and `ta_pub_key_modulus` which are not resolved, therefore change its definition to depend on the` CFG_WITH_USER_TA` flag.

### CODE REVIEW

**Comment by jforissier** [comment]
Hi @maroueneboubakri,

Thanks for the patches. Before I look further into them, would you please add one build line with `CFG_WITH_USER_TA=n` to `.shippable.yml` so that this configuration is checked? I suggest adding `_make CFG_WITH_USER_TA=n` just after `_make CFG_REE_FS_TA_BUFFERED=y`. And the same in `.azure-pipelines.yml`. One commit is enough for both changes (you can use `ci: build QEMU with CFG_WITH_USER_TA=n` as a subject for example). Thanks.

**Comment by jforissier** [comment]
For "ci: build for QEMU with CFG_WITH_USER_TA=n":

`Reviewed-by: Jerome Forissier <jerome@forissier.org>`

File: core/include/kernel/user_access.h [hunk 1]
```diff
@@ -10,25 +10,8 @@
 #include <tee_api_types.h>
 #include <types_ext.h>
 
-#ifdef CFG_WITH_USER_TA
 TEE_Result copy_from_user_private(void *kaddr, const void *uaddr, size_t len);
 TEE_Result copy_from_user(void *kaddr, const void *uaddr, size_t len);
-#else
-static inline TEE_Result copy_from_user_private(void *kaddr __unused,
```
Comment (jenswi-linaro) [inline core/include/kernel/user_access.h:17]: I'd prefer keeping these, I believe that the corresponding .c file shouldn't be built in this case.

**Comment by jenswi-linaro** [comment]
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Review by etienne-lms — COMMENTED** [review]
`Reviewed-by: Etienne Carriere <etienne.carriere@linaro.org>`

**Comment by b49020** [comment]
`Reviewed-by: Sumit Garg <sumit.garg@linaro.org>`

**Pull request merged**